fdmi/plugins/sched.c

```diff
@@ -579,46 +556,58 @@ static int sched_sighandler_init(void)
 	return rc == 0 ? 0 : M0_ERR(errno);
 }
 
-void print_params(struct sched_conf *params) {
-	printf("local: %s\n", params->local_addr);
-	printf("ha: %s\n", params->ha_addr);
-	printf("prof: %s\n", params->profile_fid);
-	printf("process: %s\n", params->process_fid);
+static void print_params(struct sched_conf *params)
+{
+	fprintf(stderr,
+		"Starting params:\n"
+		"  local_ep: %s\n"
+		"  hare_ep : %s\n"
+		"  profile_fid : %s\n"
+		"  process_fid : %s\n"
+		"  plugin_fid  : %s\n",
+		params->local_addr, params->ha_addr,
+		params->profile_fid, params->process_fid,
+		params->fdmi_plugin_fid_s);
 }
 
 int main(int argc, char **argv)
 {
 	int rc = 0;
+
 	if (argc == 1) {
-		fprintf(stderr, "Arguments are not provided.\n");
+		usage();
 		exit(EXIT_FAILURE);
 	}
+
 	rc = sched_args_parse(&c_params, argc, argv);
 	if (rc != 0) {
-		fprintf(stderr, "Sched args parse failed\n");
+		fprintf(stderr, "Args parse failed\n");
 		return M0_ERR(errno);
 	}
+
 	rc = m0_semaphore_init(&sched_sem, 0);
 	if (rc != 0)
 		return M0_ERR(errno);
 
-	//print_params(&c_params);
-
 	rc = sched_init(&c_params);
 	if (rc != 0) {
-		sched_fini(&c_params);
-		return M0_ERR(errno);
+		rc = M0_ERR(errno);
+		goto sem_fini;
 	}
 
 	rc = sched_sighandler_init();
 	if (rc != 0)
-		goto sem_fini;
-	/* main thread loop */
-	fprintf(stdout, "m0sched waiting for signal...\n");
+		goto sched_fini;
+
+        /* Main thread loop */
```

Comment:
Indentation issue.

Also, this comment is misleading.
If we say, "main thread loop", usually, we have such code:
```
        while (true) {
                 xxxxx;
        }
```
But now, the main thread is just waiting for semaphore. The main thread is blocked.

Reply:
ok

Reply:
OK